pkg/kclient/kclient.go

```diff
@@ -92,3 +107,76 @@ func CreateObjectMeta(name, namespace string, labels, annotations map[string]str
 
 	return objectMeta
 }
+
+// Delete takes labels as a input and based on it, deletes respective resource
+func (c *Client) Delete(labels map[string]string, wait bool) error {
+
+	// convert labels to selector
+	selector := util.ConvertLabelsToSelector(labels)
+	klog.V(3).Infof("Selectors used for deletion: %s", selector)
+
+	var errorList []string
+	var deletionPolicy = metav1.DeletePropagationBackground
+
+	// for --wait flag, it deletes component dependents first and then delete component
+	if wait {
+		deletionPolicy = metav1.DeletePropagationForeground
+	}
+	// Delete Deployments
+	klog.V(3).Info("Deleting Deployments")
+	err := c.appsClient.Deployments(c.Namespace).DeleteCollection(&metav1.DeleteOptions{PropagationPolicy: &deletionPolicy}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		errorList = append(errorList, "unable to delete deployments")
+	}
+
+	// for --wait it waits for component to be deleted
+	// TODO: Need to modify for `odo app delete`, currently wait flag is added only in `odo component delete`
```

Comment:
is there something remaining  `odo app delete`?

Reply:
Most of the function is copied from occlient.go. This `TODO` is also present there so I kept it.